### PR TITLE
CRS-640 hdced 4 year concurrent.

### PR DESCRIPTION
**Pull request by ldlharper**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt [hunk 1]
```diff
@@ -29,9 +29,9 @@ data class SentenceCalculation(
   var numberOfDaysToLicenceExpiryDate: Long = 0
   var licenceExpiryDate: LocalDate? = null
 
-  //  Home Detention Curfew Expiry Date(HDCED)
-  var numberOfDaysToHomeDetentionCurfewExpiryDate: Long = 0
-  var homeDetentionCurfewExpiryDateDate: LocalDate? = null
+  //  Home Detention Curfew Eligibility Date(HDCED)
+  var numberOfDaysToHomeDetentionCurfewEligibilityDate: Long = 0
+  var homeDetentionCurfewEligibilityDateDate: LocalDate? = null
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt:34]: DateDate? 😆

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt [hunk 2]
```diff
@@ -154,13 +159,21 @@ class BookingExtractionService(
...
-      booking.getAllExtractableSentences().minOf { it.sentencedAt },
+      earliestSentenceDate,
       latestUnadjustedExpiryDate
     )
 
     return bookingCalculation
   }
 
+  private fun extractManyHomeDetentionCurfewEligibilityDate(earliestSentenceDate: LocalDate, latestUnadjustedExpiryDate: LocalDate, mostRecentSentenceByReleaseDate: ExtractableSentence): LocalDate? {
+    val fourYearSentence = earliestSentenceDate.plusYears(4)
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt:170]: Might want to replace this magic numbeer at some point?

**Review by mikehalmamoj — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt (hunk 1 above)
Comment (ldlharper) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt:34]: I just really wanted to make sure people knew it was a date

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt (hunk 2 above)
Comment (ldlharper) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt]: Done

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**